src/python-common/ceph/deployment/service_spec.py

```diff
@@ -567,6 +567,12 @@ def service_name(self) -> str:
             n += '.' + self.service_id
         return n
 
+    def get_ports(self) -> List[int]:
+        return []
+
+    def get_ip(self) -> Optional[str]:
+        return None
```

Comment:
`get_ip()` can't live in ServiceSpec. This is a property of a host and not of a service spec.